src/main/java/org/variantsync/diffdetective/variation/diff/Differ.java

```diff
@@ -0,0 +1,48 @@
+package org.variantsync.diffdetective.variation.diff;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+import java.nio.file.Path;
+
+import org.variantsync.diffdetective.diff.result.DiffParseException;
+import org.variantsync.diffdetective.util.FileSource;
+import org.variantsync.diffdetective.util.Source;
+import org.variantsync.diffdetective.variation.Label;
+
+/**
+ * A generic interface for creating variation diffs from the state before and the state after an
+ * edit.
```

Comment:
And the state is considered to be text, not variation trees. So this interface represents a differ that walks the full visual abstract (i.e., the figure in our README on variability-aware differencing) from start to finish.

Maybe a better name than `Differ` would be `VariabilityAwareDiffer`. This would be consistent with the naming in my PhD thesis.